tools/RepositoryManagement/DevHome.RepositoryManagement/ViewModels/RepositoryManagementItemViewModel.cs

```diff
@@ -2,61 +2,451 @@
+    {
+        get => _branch ?? string.Empty;
+        set => _branch = value ?? string.Empty;
+    }
 
     public bool IsHiddenFromPage { get; set; }
 
+    public bool HasAConfigurationFile { get; set; }
+
     [RelayCommand]
-    public void OpenInFileExplorer()
+    public async Task OpenInFileExplorer()
     {
-        throw new NotImplementedException();
+        await CheckCloneLocationNotifyUserIfNotFound();
+        OpenRepositoryInFileExplorer(RepositoryName, ClonePath, nameof(OpenInFileExplorer));
     }
 
     [RelayCommand]
-    public void OpenInCMD()
+    public async Task OpenInCMD()
     {
-        throw new NotImplementedException();
+        await CheckCloneLocationNotifyUserIfNotFound();
+        OpenRepositoryinCMD(RepositoryName, ClonePath, nameof(OpenInCMD));
     }
 
     [RelayCommand]
-    public void MoveRepository()
+    public async Task MoveRepository()
     {
-        throw new NotImplementedException();
+        // TODO: Save to the database before moving the folder.
+        var newLocation = await PickNewLocationForRepositoryAsync();
+
+        if (string.IsNullOrEmpty(newLocation))
+        {
+            _log.Information("The path from the folder picker is either null or empty.  Not updating the clone path");
+            return;
+        }
+
+        if (string.Equals(Path.GetFullPath(newLocation), Path.GetFullPath(ClonePath), StringComparison.OrdinalIgnoreCase))
+        {
+            _log.Information("The selected path is the same as the current path.  Not updating the clone path");
+            return;
+        }
+
+        var newDirectoryInfo = new DirectoryInfo(Path.Join(newLocation, RepositoryName));
+        var currentDirectoryInfo = new DirectoryInfo(Path.GetFullPath(ClonePath));
+
+        try
+        {
+            currentDirectoryInfo.MoveTo(newDirectoryInfo.FullName);
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, $"Cound not move repository to the selected location.");
+            TelemetryFactory.Get<ITelemetry>().Log(
+                EventName,
+                LogLevel.Critical,
+                new RepositoryLineItemEvent(nameof(MoveRepository), RepositoryName));
+        }
+
+        var repository = GetRepositoryReportIfNull(nameof(MoveRepository));
+        if (repository == null)
+        {
+            return;
+        }
+
+        var didUpdate = _dataAccess.UpdateCloneLocation(repository, newDirectoryInfo.FullName);
+
+        if (!didUpdate)
+        {
+            _log.Warning($"Could not update the database.  Check logs");
+        }
+
+        ClonePath = Path.Join(newLocation, RepositoryName);
     }
 
     [RelayCommand]
-    public void DeleteRepository()
+    public async Task DeleteRepository()
     {
-        throw new NotImplementedException();
+        var cantFindRepositoryDialog = new ContentDialog()
+        {
+            XamlRoot = _window.Content.XamlRoot,
+            Title = $"Would you like to delete this repository?",
```

Comment:
When you're updating the strings, I think it would be a wise choice to make the user have to type in the name of the repository before allowing them to delete. Also, we should have the repository name in the dialog header as well and probably put the location on disk that we're deleting as well in the dialog. This way they'll have to be fully aware of what they're about to do.

Reply:
Added as a TODO.  Thank you.